oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/RateLimitUtils.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jackrabbit.oak.plugins.index;
+
+import java.lang.management.ManagementFactory;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Calendar;
+import java.util.Set;
+
+import javax.management.MBeanServerConnection;
+import javax.management.ObjectName;
+
+import org.apache.jackrabbit.util.ISO8601;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RateLimitUtils {
+    
+    private final static Logger LOG = LoggerFactory.getLogger(RateLimitUtils.class);    
+
+    // when getOldestAsyncIndexUpdate was called the last time 
+    private static Instant lastJmxBeanCall = Instant.now();
+
+    // the return value of the last call to getOldestAsyncIndexUpdate
+    private static Instant cachedOldestAsyncUpdate = Instant.now();
+    
+    // the last time the rateLimitWrites was called
+    private static Instant lastRateLimitCall = Instant.now();
```

Comment:
Should all these static vars be uppercased?

Reply:
+1

Reply:
They are not constants, they are only static... According to the Google style sheet, only constants should be all uppercase: https://stackoverflow.com/questions/1417190/should-a-static-final-logger-be-declared-in-upper-case